### PR TITLE
3-GET/api/topics

### PR DESCRIPTION
**Pull request by Adedunsola**
#3. GET /api/topics

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        .expect(200)
+        .then(({body: {topics}})=>{
+            expect(topics).toBeInstanceOf(Array);
+            expect(topics).toHaveLength(3);
+            topics.forEach((topic)=>{
+                expect(topic).toEqual(
+                expect.objectContaining({
+                  description: expect.any(String),
+                  slug: expect.any(String), 
+                })
```
Comment (leekli) [inline __tests__/app.test.js:48]: Good test, your checking the structure and shape of the data returned to ensure it fits the structure you're expecting

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                )
+            })
+        });
+    });
+    test('404, returns Not Found when invalid API is given', ()=>{
+        return request(app)
+        .get('/api/banana')
+        .expect(404)
+        .then(({body: {msg}})=>{
+            expect(msg).toBe('Not Found');
```
Comment (leekli) [inline __tests__/app.test.js:58]: Brill, good to see you testing for some errors

File: app.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+app.get('/api/topics', getTopics);
+
+
+
+
+//Error Handling
+app.all('*', manage404Errors);
+app.use((err,req,res,next)=>{
+    console.log(err);
+    res.sendStatus(500);
```
Comment (leekli) [inline app.js]: You probably don't need a console log here as you're dealing with the response by sending a response to the user, I'd recommend removing this console log.

File: controllers/errors.controller.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// const manageCustomError = (err,req,res,next) => {
+//     if(err.msg && err.status){
+//         res.staus(err.status).send({msg: err.msg})
+//     }else{
+//         next(err)
+//     }
+// };
```
Comment (leekli) [inline controllers/errors.controller.js]: I shouldn't see commented out code as part of a commit which is meant to be production-ready, please remove this

File: controllers/topics.controller.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+const {selectTopics} = require('../models/topics.model');
+
+
+exports.getMessage = (req,res,next)=>{
+    const msg = 'OK';
+    return res.status(200).send(msg);
+};
+
+exports.getTopics = (req,res,next)=>{
+    selectTopics()
```
Comment (leekli) [inline controllers/topics.controller.js:10]: Good naming conventions for your controller and model - well done!

File: controllers/topics.controller.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    .then((topics)=>{
+        res.status(200).send({topics});
```
Comment (leekli) [inline controllers/topics.controller.js:12]: response being sent back on a keyword of 'topics' is spot on, great work

**Review by leekli — CHANGES_REQUESTED** [review]
Hi Ade,

Great work on this ticket. I have left comments around the commit for you to read.

It's not quite ready to be merged yet, there are things like a console log, commented out code which need removing to make this commit 'production ready'.

Please review these comments, make the necessary changes and then submit a new pull request for review. Please do not merge this branch until we have approved it after the next pull request with the requested changes made.

**Review by RichardNeat — COMMENTED** [review]
Great merge away :)

**Pull request merged**